### PR TITLE
Gang Update v3

### PR DESCRIPTION
**Pull request by Ikarrus**
tldr; Gang mode rewrite to follow my original vision: Longer, more deliberate and meaningful rounds with less directionless deathmatching, driven by a resource both teams have to fight over.

NEW OBJECTIVE: CLAIM HALF THE STATION AS TERRITORY
- Gang bosses can spawn special spray cans that can be used to tag areas
- Any gang member can tag areas.
- Gang tags must be removed before another one is placed in the same area.
- Alternatively, you can spray your tag over an opponent's and it will replace it
- Game mode no longer automatically ends when the boss dies. It now ends when either one gang claims more than 50% of the station areas, or the station evacuates.

NEW RECRUITMENT TOOL: PENS
- Recruitment pens replace flashes
- Looks exactly like a pen. Stab people to recruit them.
- Recruitment is silent but obvious to the perceptive (Causes a brief seizure)
- Cooldown between uses that gets longer as the gang gets larger

NEW TOOL: GANGTOOL
- Replaces the recaller device and traitor uplink. Can still recall the shuttle.
- Only usable by the gang boss and his lieutenants
- Can purchase pistols and ammo
- Can purchase additional territory spraycans
- Can purchase additional recruitment pens
- Can purchase additional gangtools (For personal use, or promoting a single ganger to lieutenant)
- Lieutenants can do anything bosses can except promote others to lieutenant
- Lieutenants are immune to deconversions
- Notifies user of promotions, purchases, new territory, and lost territory within the gang.

NEW CURRENCY SYSTEM: INFLUENCE
- Gangtools purchase items with influence accrued over time
- Influence is gained every 5 minutes. The more territory a gang has claimed, the more influence they get.
- Influece are accumulated per-gang, so having more gangtools doesn't mean you get more influence

MISCELLANEOUS CHANGES
- Gang no longer supports continuous rounds
- Default gang probability set to 2 (I also dropped monkey to 0 because it's broken)
- Fixed janitors not being able to clean walls
- Janitors can clean thermite off walls now
- Fixed some bugs with spraycans

### CODE REVIEW

**Comment by Cheridan** [comment]
Name of gang currency should be "Swag"

**Comment by Alek2ander** [comment]
Maybe change the income to be dependent on the size of the areas claimed? I'd say Central Primary Hallway would be worth a lot more than some 2x3 security office.

**Comment by Dorsisdwarf** [comment]
Is there a cap on lieutenants?

**Comment by hanshansenhansson** [comment]
>Gang tags must be removed before another one is placed in the same area

I think it would be better to have everyone tag the area, and whoever has most tags in the end gets it. Else it becomes a game of "hide the tag".

**Comment by Miauw62** [comment]
I would like to note that it is currently impossible for the janitor to clean graffiti off of walls.

**Comment by Miauw62** [comment]
also there should probably be a reason not to promote everyone to lieutenant as soon as you get the points.

**Comment by Steelpoint** [comment]
A limit of one or two lieutenant's would be a good limit, also maybe make it take a while to promote a replacement.

I agree that at the very least the Janitor should be able to remove graffiti. That or something in Sec to act as a counter to gang activity.

Also it would be interesting to look at making some areas of the ship give more Swag every 5 minutes, say the Captain's Office or Armoury would give a very high amount of resources, but some hallway would provide little.

**Comment by JJRcop** [comment]
Have you thought about making gang mulligan if it takes too long or the gang is killed off?

**Comment by hanshansenhansson** [comment]
Downside of more lieutenants is that you have spent points you could have used for real weapons or converter pens. I don't think they should need a hard limit, the boss will most likely not make too many of them as every one of them uses the same resource pool and you do not want fools to waste your swag on bling.

**Comment by JJRcop** [comment]
What do you think about making gang mulligan if a boss dies?

**Comment by Ikarrus** [comment]
Lieutenants are soft limited by their high cost and risk due to everyone buying from the same pool of funds. If it becomes an issue I can cap it later, but I'd like to see how this rolls out first.

The idea is that points are hard to come by. You shouldn't be able to mass-produce weapons or gangtools. Keep in mind that you will likely need to constantly spend points to buy territory spraycans to give to your gang, which have limited uses each. But if it reaches that point, chances are you already control the station.

**Comment by Ikarrus** [comment]
>I think it would be better to have everyone tag the area, and whoever has most tags in the end gets it. Else it becomes a game of "hide the tag". Especially in large areas like the central hallway

That was sort of the idea. The tags would be something that would need to be protected, a "control point" for the area. I'm also likely to make maint areas invalid for territory marking.

**Comment by Iamgoofball** [comment]
You should make it randomize the gang name so we can have TUNNEL CLOWNS vs SCIENCE BLASTERS

**Comment by Ikarrus** [comment]
@JJRcop I don't think it would be fair to either team if new antags were spawned if there are any gangers left. They can still win without a boss. That said, it's very unlikely that all gangsters will be eliminated from the game.

Also I'm not using Swag.

**Comment by Ikarrus** [comment]
@Iamgoofball confirmed to not play the game

**Comment by Iamgoofball** [comment]
>wanting to play the game as a coder 

Also :+1:

**Comment by Dorsisdwarf** [comment]
Removing all gangsters of both teams from play should result in a minor crew victory. 
If all heads and all gangsters are dead, it should declare a draw.

**Comment by Ikarrus** [comment]
I'm just saying that it's an unnecessarily expensive operation for such an edge case.

If every ganger is dead, then the crew can still play the game normally and evacuate whenever they want.

**Comment by Cheridan** [comment]
>Also I'm not using Swag.

:broken_heart:

**Comment by RemieRichards** [comment]
Call it "influence"

**Comment by Lumipharon** [comment]
>not 'street cred'

If that hotline miami weapon pr ever gets/got merged, let the tool spawn some stuff from that.
Also an easier way to get all your people into gang colours would be neat.

Hiding your controlling tag thing is bad, but 'flood the room with as many tags as possible to win' is even worse. Need to figure out a way to make it obvious (atleast to gangsters) that a room is tagged, and where it is.

**Comment by Iamgoofball** [comment]
@Cheridan when this is merged I can just make a future PR to randomize the name so it can be "swag", "influence", "cash", "dosh", "points", whatever

**Comment by RemieRichards** [comment]
Sex appeal.
You have earned 500 Sex appeal!

**Comment by Iamgoofball** [comment]
exactly

**Comment by Ikarrus** [comment]
>Goofball touching this

Motivation lost.

**Comment by Iamgoofball** [comment]
Gang Mania 1: Reagent Based Bosses

**Comment by Ikarrus** [comment]
It's been updated, it's been tested, and AFAIK it's working! It's good to go now!

**Comment by RemieRichards** [comment]
>it's called influence

YAY, I WIN <3 @Ikarrus

**Comment by Cheridan** [comment]
i'm cring
